lib/class-wp-json-comments-controller.php

```diff
@@ -0,0 +1,470 @@
+			'comment_date'         => isset( $request['date'] ) ? $request['date'] : current_time( 'mysql' ),
+			'comment_date_gmt'     => isset( $request['date_gmt'] ) ? $request['date_gmt'] : current_time( 'mysql', 1 ),
+			// Setting remaining values before wp_insert_comment so we can
+			// use wp_allow_comment().
+			'comment_author_IP'    => '127.0.0.1',
+			'comment_agent'        => '',
+		);
+
+		return apply_filters( 'json_preprocess_comment', $prepared_comment, $request );
+	}
+
+	/**
+	 * Prepare a single comment for database update.
+	 *
+	 * @param  WP_JSON_Request $request Request object.
+	 * @return array           $prepared_comment
+	 */
+	protected function prepare_item_for_update( $request ) {
+		$prepared_comment = array();
+
+		if ( isset( $request['content'] ) ) {
+			$prepared_comment['comment_content'] = $request['content'];
+		}
+
+		if ( isset( $request['author'] ) ) {
+			$prepared_comment['comment_author'] = sanitize_text_field( $request['author'] );
+		}
+
+		if ( isset( $request['author_email'] ) ) {
+			$prepared_comment['comment_author_email'] = sanitize_email( $request['author_email'] );
+		}
+
+		if ( isset( $request['author_url'] ) ) {
+			$prepared_comment['comment_author_url'] = esc_url_raw( $request['author_url'] );
+		}
+
+		if ( ! empty( $request['date'] ) ) {
+			$prepared_comment['comment_date'] = $request['date'];
+		}
+
+		return apply_filters( 'json_preprocess_comment', $prepared_comment, $request );
+	}
+
+	/**
+	 * Process a comment_status change when updating a comment.
+	 *
+	 * @param string|int $new_status
+	 * @param object     $comment
+	 * @return boolean   $changed
+	 */
+	protected function handle_status_change( $new_status, $comment ) {
+		$old_status = wp_get_comment_status( $comment->comment_ID );
+
+		if ( $new_status == $old_status ) {
+			return false;
+		}
+
+		switch ( $new_status ) {
+			case 'approved' :
+			case 'approve':
+			case '1':
+				$changed = wp_set_comment_status( $comment->comment_ID, 'approve' );
+				break;
+			case 'hold':
+			case '0':
+				$changed = wp_set_comment_status( $comment->comment_ID, 'hold' );
+				break;
+			case 'spam' :
+				$changed = wp_spam_comment( $comment->comment_ID );
+				break;
+			case 'unspam' :
+				$changed = wp_unspam_comment( $comment->comment_ID );
+				break;
+			case 'trash' :
+				$changed = wp_trash_comment( $comment->comment_ID );
+				break;
+			case 'untrash' :
+				$changed = wp_untrash_comment( $comment->comment_ID );
+				break;
+			default :
+				$changed = false;
+				break;
+		}
+
+		return $changed;
+	}
+
+	/**
+	 * Check if we can read a post.
+	 *
+	 * Correctly handles posts with the inherit status.
+	 *
+	 * @param  WP_Post $post Post Object.
+	 * @return boolean Can we read it?
+	 */
+	protected function check_read_post_permission( $post ) {
+		$posts_controller = new WP_JSON_Posts_Controller;
+
+		return $posts_controller->check_read_permission( $post );
```

Comment:
Seems too tightly-coupled. Can we do it in a more de-coupled manner?

Do comments always need to have a post? If so, why are we exposing them at `/wp/comments` ?